splitgpg2/__init__.py

```diff
@@ -629,8 +632,9 @@ async def command_SETHASH(self, untrusted_args: Optional[bytes]):
 
         if not untrusted_hash:
             raise Filtered
-
-        hash_regex = re.compile(rb'\A[0-9A-F]{%d}\Z' % alg_param.len)
+        if untrusted_hash.len() != alg_param.len:
+            raise Filtered
+        hash_regex = re.compile(rb'\A[0-9A-F]+\Z')
```

Comment:
Not a fan of splitting it. As soon as the regex would be even a little bit more complicated (for example a non hex prefix) the split check would get much harder. Did you really noticed the speed difference?